storage/src/test/java/tech/pegasys/artemis/storage/server/AbstractStorageBackedDatabaseTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.storage.server;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.io.Files;
+import com.google.common.primitives.UnsignedLong;
+import java.io.File;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.tuweni.bytes.Bytes32;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import tech.pegasys.artemis.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.artemis.datastructures.state.BeaconState;
+import tech.pegasys.artemis.datastructures.state.Checkpoint;
+import tech.pegasys.artemis.storage.TrackingStorageUpdateChannel;
+
+public abstract class AbstractStorageBackedDatabaseTest extends AbstractDatabaseTest {
+
+  protected abstract Database createDatabase(
+      final File tempDir, final StateStorageMode storageMode);
+
+  @Override
+  protected Database createDatabase(final StateStorageMode storageMode) {
+    final File tmpDir = Files.createTempDir();
+    tmpDir.deleteOnExit();
```

Comment:
Does this work for a directory given it won't be empty by the time the test finishes?

Reply:
Good catch - I've added extra cleanup logic for this.